### PR TITLE
Seanmcm/windows sdk version

### PR DESCRIPTION
**Pull request by sean-mcmanus**
I'm still reviewing/testing this some more, so you could wait a while if you want...

### CODE REVIEW

File: Extension/package.json [hunk 1]
```diff
@@ -286,6 +286,15 @@
           "description": "The value to use in a configuration if \"macFrameworkPath\" is not specified, or the values to insert if \"${default}\" is present in \"macFrameworkPath\".",
           "scope": "resource"
         },
+        "C_Cpp.default.windowsSDKVersion": {
```
Comment (bobbrow) [inline Extension/package.json]: `SDK` should be `Sdk` per convention

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/c_cpp_properties.schema.json [hunk 2]
```diff
@@ -58,6 +58,10 @@
                             "type": "string"
                         }
                     },
+                    "windowsSDKVersion": {
+                        "description": "Version of the Windows SDK include path to use on Windows, e.g. '10.0.17134.0'.",
+                        "type": "string"
```
Comment (bobbrow) [inline Extension/c_cpp_properties.schema.json]: should probably add `patternProperties` ([reference](https://spacetelescope.github.io/understanding-json-schema/reference/object.html)) to this so we can get squiggles for invalid versions.  (see the `env` property).

The regexp is this, I think: `^\d{2}\.\d{1}\.\d{5}\.\d{1}$`

File: Extension/package.json (hunk 1 above)
Comment (sean-mcmanus) [reply Extension/package.json]: Ah, okay -- that's easier to read :)

File: Extension/c_cpp_properties.schema.json (hunk 2 above)
Comment (sean-mcmanus) [reply Extension/c_cpp_properties.schema.json]: Ok.

File: Extension/c_cpp_properties.schema.json [hunk 3]
```diff
@@ -58,6 +58,11 @@
                             "type": "string"
                         }
                     },
+                    "windowsSdkVersion": {
+                        "description": "Version of the Windows SDK include path to use on Windows, e.g. '10.0.17134.0'.",
+                        "type": "string",
+                        "pattern": "^\\d{2}\\.\\d{1}\\.\\d{5}\\.\\d{1}$|^8.1$"
```
Comment (bobbrow) [inline Extension/c_cpp_properties.schema.json]: The 8.1 part should have a backslash before the dot

File: Extension/package.json [hunk 4]
```diff
@@ -286,6 +286,15 @@
...
           "scope": "resource"
         },
+        "C_Cpp.default.windowsSdkVersion": {
+          "type": [
+            "string",
+            "null"
+          ],
+          "default": null,
+          "description": "Version of the Windows SDK include path to use on Windows, e.g. '10.0.17134.0'.",
+          "scope": "resource"
```
Comment (bobbrow) [inline Extension/package.json]: Should the pattern go in here too?

**Review by bobbrow — APPROVED** [review]

File: Extension/c_cpp_properties.schema.json (hunk 3 above)
Comment (sean-mcmanus) [reply Extension/c_cpp_properties.schema.json]: Ooops...

File: Extension/package.json (hunk 4 above)
Comment (sean-mcmanus) [reply Extension/package.json]: Hmm...it doesn't seem to be required...I could try adding it...oh, this is the default one...I didn't test that yet...

**Review by bobbrow — APPROVED** [review]

**Pull request merged**